### ISSUE TITLE
docker stays at "Starting Kodi" - Omega version

### ISSUE THREAD

**Issue by Bumblebrow** [matthuisman/docker-kodi-headless#60]
I have tried installing this docker on a Debian server and Ubuntu server, but the results are the same. The logs don't reflect any errors, but it just stays at "Starting Kodi.." I tested using different OID and GID, off box SQL, no SQL, etc. but the results are the same. When I look in the kodi.log, it shows no errors as well. The last thing entry in the kodi.log reflects the Addons33.db being opened and then uninstalling/installing addons. Nothing after that. Is there anything else I can check to see why it's hanging without generating an error of any kind?

kodi-headless yml:
kodi-headless:
  image: matthuisman/kodi-headless:Omega
  container_name: kodi-headless
  environment:
    - PUID=0
    - PGID=0
    - MYSQL_HOST=*****
    - MYSQL_USER=*****
    - MYSQL_PASSWORD=****
    - KODI_DELAY=30
    - TZ=America/Chicago
  volumes:
    - /mnt/docker-data/:/config/.kodi
  ports:
    - 8080:8080
    - 9090:9090
    - 9777:9777/udp
  restart: always

docker logs:
───────────────────────────────────────
  _____ __ __ _____ _____ _____ _____
 |     |  |  |   __|_   _|     |     |
 |   --|  |  |__   | | | |  |  | | | |
 |_____|_____|_____| |_| |_____|_|_|_|
       _____ __ __ _ __    ____
      | __  |  |  | |  |  |    \
      | __ -|  |  | |  |__|  |  |
      |_____|_____|_|_____|____/

  Based on images from linuxserver.io
───────────────────────────────────────

To support the app dev(s) visit:
Matt Huisman: https://www.matthuisman.nz/donate
Kodi: https://kodi.tv/donate

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    0
User GID:    0
───────────────────────────────────────

Setting timezone to America/Chicago...
[custom-init] No custom files found, skipping...
Waiting for 30s before starting Kodi..
[ls.io-init] done.
Starting Kodi..

kodi.log:
2024-11-14 09:35:16.339 T:182      info <general>: creating subdirectories
2024-11-14 09:35:16.339 T:182      info <general>: userdata folder: special://masterprofile/
2024-11-14 09:35:16.339 T:182      info <general>: recording folder: 
2024-11-14 09:35:16.339 T:182      info <general>: screenshots folder: 
2024-11-14 09:35:16.392 T:182      info <general>: Running database version Addons33
2024-11-14 09:35:16.399 T:182     debug <general>: CAddonInfoBuilder::ParseXMLTypes: Binary addon found: screensaver.xbmc.builtin.dim
2024-11-14 09:35:16.402 T:182     debug <general>: CAddonInfoBuilder::ParseXMLTypes: Binary addon found: vfs.rar
2024-11-14 09:35:16.404 T:182     debug <general>: CAddonInfoBuilder::ParseXMLTypes: Binary addon found: vfs.sftp
2024-11-14 09:35:16.410 T:182     debug <general>: CAddonInfoBuilder::ParseXMLTypes: Binary addon found: audioencoder.kodi.builtin.aac
2024-11-14 09:35:16.412 T:182     debug <general>: CAddonInfoBuilder::ParseXMLTypes: Binary addon found: vfs.libarchive
2024-11-14 09:35:16.412 T:182     debug <general>: CAddonInfoBuilder::ParseXMLTypes: Binary addon found: screensaver.xbmc.builtin.black
2024-11-14 09:35:16.414 T:182     debug <general>: CAddonInfoBuilder::ParseXMLTypes: Binary addon found: audioencoder.kodi.builtin.wma
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: inputstream.adaptive has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: inputstream.ffmpegdirect has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: inputstream.rtmp has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: peripheral.joystick has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: pvr.hdhomerun has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: pvr.iptvsimple has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.certifi has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.chardet has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.dateutil has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.dropbox has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.future has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.idna has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.pathtools has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.pyqrcode has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.requests has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.six has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.urllib3 has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.module.watchdog has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: script.xbmcbackup has been uninstalled.
2024-11-14 09:35:16.420 T:182     debug <general>: CAddonDatabase: service.librarywatchdog has been uninstalled.

**Comment by matthuisman** [comment]
No ideas sorry. Kodi is starting as it's logging. 
I'd suggest starting with fresh user data. Seems you've got a bunch of addon's or something

**Comment by Bumblebrow** [comment]
I originally tried with new user data but it fails with an Addon33.db creation. I took the Addons33.db from another machine and it at least got passed the issue of starting with no userdata at the beginning. I deleted all data and started from scratch again with just the docker.yml and this is the results.

from docker:

Setting timezone to America/Chicago...
[custom-init] No custom files found, skipping...
Waiting for 30s before starting Kodi..
[ls.io-init] done.
Starting Kodi..
terminate called after throwing an instance of 'dbiplus::DbErrors'
Aborted (core dumped)
Crash report available at /config/kodi_crashlog-20241114_140137.log
Kodi finished
Waiting for 30s before starting Kodi..
Starting Kodi..
terminate called after throwing an instance of 'dbiplus::DbErrors'
Aborted (core dumped)
Crash report available at /config/kodi_crashlog-20241114_140209.log
Kodi finished
Waiting for 30s before starting Kodi..

kodi.log:
2024-11-14 14:01:37.694 T:182   warning <CSettingsManager>: missing version attribute
2024-11-14 14:01:37.695 T:182      info <general>: Default Video Player: VideoPlayer
2024-11-14 14:01:37.695 T:182      info <general>: Default Audio Player: paplayer
2024-11-14 14:01:37.695 T:182      info <general>: Disabled debug logging due to GUI setting. Level 1.
2024-11-14 14:01:37.705 T:182     debug <general>: CSkinSettings: no <skinsettings> tag found
2024-11-14 14:01:37.705 T:182      info <general>: New Cache GUI Settings (replacement of cache in advancedsettings.xml) are:
                                                     Buffer Mode: 4
                                                     Memory Size: 20 MB
                                                     Read Factor: 4.00 x 
                                                     Chunk Size : 131072 bytes
2024-11-14 14:01:37.705 T:182      info <general>: creating subdirectories
2024-11-14 14:01:37.705 T:182      info <general>: userdata folder: special://masterprofile/
2024-11-14 14:01:37.705 T:182      info <general>: recording folder: 
2024-11-14 14:01:37.705 T:182      info <general>: screenshots folder: 
2024-11-14 14:01:37.756 T:182     error <general>: SQL: [Addons33.db] SQLite error SQLITE_ERROR (no such table: version)
                                                   Query: SELECT idVersion FROM version

**Comment by matthuisman** [comment]
That tells me the addon's.db isn't good. Is that first log with no db file?
You sure your starting with fresh user data?

**Comment by matthuisman** [comment]
Try without mounting any volume to .kodi

**Comment by Bumblebrow** [comment]
Correct, the last one was without and .kodi folder pre-existing. initially, when docker started, a userdata folder appeared for 30 seconds until kodi started. Once kodi started, all other folders were created and a 0kb addon33.db.

I'll remove the mount volume as you suggested and report back.

**Comment by Bumblebrow** [comment]
Thanks! Putting the config mount point back to the host of the docker worked!

**Comment by matthuisman** [comment]
Id suggest it was a permission issue to your mount point. From your config,
you probably should have used another directory inside your docker volume.
And make sure docker had correct permission

**Comment by Bumblebrow** [comment]
I agree. Although root was owner and group and 0 was passed, it would fail every time after creating the 0kb addon33.db. Once I tested your suggestion, it created the structure, the files went through, and everything is working. I moved that folder to the mount point again and it is still working. Something with the initial creation of the kodi addon33.db was failing but all is good now!. Thanks for your help!
